provisioning/miniprov/hare_mp/utils.py

```diff
@@ -160,12 +160,14 @@ def save_log_path(self):
         self.kv.kv_put(f'{hostname}/log_path', log_path)
 
     @func_log(func_enter, func_leave)
-    def is_motr_component(self, machine_id: str) -> bool:
+    def is_motr_io_present(self, machine_id: str) -> bool:
         """
         Returns True if motr component is present in the components list
         for the given node>{machine_id}.
         """
-        return self.is_component(machine_id, Const.COMPONENT_MOTR.value)
+        return self.is_component_and_service(machine_id,
```

Comment:
How about,
- get_components() -> List
- get_component_services -> List
- have a dictionary and corresponding function, e.g.
   - component_build_map = { Const.COMPONENT_MOTR.value : build_motr}
- every build_<component> function must fetch list of services, if the list is empty use default services for that component (this also must be predefined). e.g. build_motr(self, machine-id: str)
- for each component in components
    - self.component_build_map[component](machine-id)
- def build_motr():
        services = get_motr_services(machine_id)
        if not services:
            build_motr_default(machine_id)
        else:
            build_motr_with_services(machine_id, services)

Note that this is rough implementation, you may need to improvise if required but overall this way we can avoid some spaghetti code and keep implementation more modular to extend easily later.

Reply:
@mssawant how big is this change? can we create subsequent ticket and start working immediately after merging this PR ? cc @SwapnilGaonkar7 @SunilSavanurSeagate @supriyachavan4398

Reply:
OKay, let's create a separate ticket for improvisation, I think we also need one for unit tests, please create both the tickets and update the PR.

Reply:
@mssawant  new tickets are created for further improvment and unit tests. Mentioned same in other comment by @supriyachavan4398

Reply:
@mssawant Created a new Jira at https://jts.seagate.com/browse/CORTX-30763, https://jts.seagate.com/browse/CORTX-30764 for the above comments.
cc. @vaibhavparatwar, @SwapnilGaonkar7, @Shreya-18